src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs

```
@@ -112,27 +113,91 @@ private void WorkerThread()
-                            var metricValueBuilder = builder.AddValue();
-                            metricValueBuilder = metricValueBuilder.WithValue(value);
-                            metricValueBuilder.WithLabel(label.Key, label.Value);
+                            foreach (var metric in this.exporter.GetAndClearDoubleMetrics())
+                            {
+                                var labels = metric.Labels;
+                                var builder = new PrometheusMetricBuilder()
+                                    .WithName(metric.MetricName)
+                                    .WithDescription(metric.MetricDescription);
+
+                                switch (metric.AggregationType)
+                                {
+                                    case AggregationType.DOUBLESUM:
+                                        {
+                                            var doubleSum = metric.Data as SumData<double>;
+                                            var doubleValue = doubleSum.Sum;
+
+                                            builder = builder.WithType("counter");
+
+                                            foreach (var label in labels)
+                                            {
+                                                var metricValueBuilder = builder.AddValue();
+                                                metricValueBuilder = metricValueBuilder.WithValue(doubleValue);
+                                                metricValueBuilder.WithLabel(label.Key, label.Value);
+                                            }
+
+                                            builder.Write(writer);
+                                            break;
+                                        }
+
+                                    case AggregationType.LONGSUM:
+                                        {
+                                            // This cannot occcur as we are iterating Double metrics.
```

Please report an error here